elektronLogging/clsfnTaskDistOverheadLogger.go

```diff
@@ -0,0 +1,50 @@
+package elektronLogging
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"strings"
+)
+
+type ClsfnTaskDistOverheadLogger struct {
+	LoggerImpl
+}
+
+func NewClsfnTaskDistOverheadLogger(logType int, prefix string) *ClsfnTaskDistOverheadLogger {
+	cLog := &ClsfnTaskDistOverheadLogger{}
+	cLog.Type = logType
+	cLog.SetLogFile(prefix)
+	return cLog
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) Log(logType int, level log.Level, logData log.Fields, message string) {
+	if cLog.Type == logType {
+
+		logger.SetLevel(level)
+
+		if cLog.AllowOnConsole {
+			logger.SetOutput(os.Stdout)
+			logger.WithFields(logData).Println(message)
+		}
+
+		logger.SetOutput(cLog.LogFileName)
+		logger.WithFields(logData).Println(message)
+	}
+	if cLog.next != nil {
+		cLog.next.Log(logType, level, logData, message)
+	}
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) SetLogFile(prefix string) {
```

Comment:
You can possible rewrite this function to something like the one shown below.
```go
func (cLog *ClsfnTaskDistOverheadLogger) SetLogFile(prefix string) {
    filename := prefix + config.TaskDistConfig.FilenameExtension
    if logDir != "" {
        if logFile, err := os.Create(filepath.Join(logDir, filename)); err != nil {
            log.Fatal("Unable to create logFile: ", err)
        } else {
            cLog.LogFileName = logFile
            cLog.AllowOnConsole = config.TaskDistConfig.AllowOnConsole
        }
    }
}
```

Reply:
Done